packages/core/src/utils/getActivityLivestreamingMetadata.ts

```diff
@@ -14,74 +14,106 @@ import {
+  streamId: pipe(string(), nonEmpty()),
+  streamType: literal('final')
+});
+
+const channelDataStreamingActivitySchema = union([
+  // Interim or Informative message
+  // Informative may not have "text", but should have abstract instead (checked later)
+  object({
+    type: literal('typing'),
+    channelData: activeSchema,
+    entities: optional(array(any()), EMPTY_ARRAY),
+    ...activityExtras
+  }),
+  // Conclude with a message.
   object({
-    attachments: optional(array(any()), EMPTY_ARRAY),
-    channelData: object({
-      // "streamId" is optional for the very first activity in the session.
-      streamId: optional(undefinedable(string())),
-      streamSequence: streamSequenceSchema,
-      streamType: literal('streaming')
-    }),
-    id: string(),
-    // "text" is optional. If not set or empty, it presents a contentless activity.
-    text: optional(undefinedable(string())),
-    type: literal('typing')
+    // If "text" is empty, it represents "regretting" the livestream.
+    type: literal('message'),
+    channelData: finalActivitySchema,
+    entities: optional(array(any()), EMPTY_ARRAY),
+    ...activityExtras
   }),
-  // Informative message.
+  // Conclude without a message.
   object({
-    attachments: optional(array(any()), EMPTY_ARRAY),
-    channelData: object({
-      // "streamId" is optional for the very first activity in the session.
-      streamId: optional(undefinedable(string())),
-      streamSequence: streamSequenceSchema,
-      streamType: literal('informative')
-    }),
-    id: string(),
-    // Informative may not have "text", but should have abstract instead (checked later)
-    text: optional(undefinedable(string())),
+    // If "text" is not set or empty, it represents "regretting" the livestream.
     type: literal('typing'),
-    entities: optional(array(any()), EMPTY_ARRAY)
+    channelData: finalActivitySchema,
+    entities: optional(array(any()), EMPTY_ARRAY),
+    text: optional(undefinedable(literal(''))),
+    ...activityExtrasFinal
+  })
+]);
+
+const entitiesStreamingActivitySchema = union([
+  // Interim or Informative message
+  // Informative may not have "text", but should have abstract instead (checked later)
+  object({
+    type: literal('typing'),
+    entities: array(activeSchema),
+    channelData: optional(any()),
+    ...activityExtras
   }),
   // Conclude with a message.
   object({
-    attachments: optional(array(any()), EMPTY_ARRAY),
-    channelData: object({
-      // "streamId" is required for the final activity in the session.
-      // The final activity must not be the sole activity in the session.
-      streamId: pipe(string(), nonEmpty()),
-      streamType: literal('final')
-    }),
-    id: string(),
     // If "text" is empty, it represents "regretting" the livestream.
-    text: optional(undefinedable(string())),
-    type: literal('message')
+    type: literal('message'),
+    entities: array(finalActivitySchema),
```

Comment:
Few things:

- `type` is required for everything under `entities` array
- In the `entities` array, there are an existing entity that describe the message itself, consider putting the livestreaming metadata in it. This is because the "livestreaming metadata" is **related to the message**
  - Otherwise, if a separate entity object is preferred, please associate it with the message entity, so people look at the graph would know "this livestreaming data is talking about the message"